### PR TITLE
Added the developer guide to writing a new connector.

### PR DESCRIPTION
**Pull request by simlay**
Rendered: 
image

### CODE REVIEW

**Comment by ajhunyady** [comment]
A couple of suggestions:
* describe the sections in the connector configuration file.
* what's the syntax used to generate this configuration file. Is it https://json-schema.org/specification.html ?

File: content/connectors/developer-guide/overview.md [hunk 1]
```diff
@@ -2,3 +2,99 @@
...
+schema](https://json-schema.org/) of the commandline arguments. This is the
+command we use to build our connectors library and validate arguments passed to
+a connector.
+
+This `metadata` subcommand should print to stdout something of the following:
+```json
+{
+    "name": "test-connector",
+    "direction": "Source",
+    "schema": {
```
Comment (nicholastmosher) [inline content/connectors/developer-guide/overview.md:35]: I would recommend showing the command that printed this output, e.g.

```
$ test-connector metadata
{ ... }
```

that way people have more context on the fact that it's a subcommand to their connector executable

File: content/connectors/developer-guide/overview.md [hunk 2]
```diff
@@ -1,5 +1,139 @@
...
-title: Overview
+title: Developer Guide
 weight: 10
-hidden: true
+hidden: false
 ---
+
+Writing a new connector is meant to be easy. You may write your connector in
+any language which has a [fluvio client library](/api). You should simply
+follow a few conventions and will gladly add and maintain a connector in our
```
Comment (nicholastmosher) [inline content/connectors/developer-guide/overview.md:9]: `s/and will gladly/and we will gladly/`

File: content/connectors/developer-guide/overview.md [hunk 3]
```diff
@@ -1,5 +1,139 @@
...
+official catalog.
+
+To do so you'll need to following steps:
```
Comment (nicholastmosher) [inline content/connectors/developer-guide/overview.md:12]: `s/need to following/need to do the following/`

File: content/connectors/developer-guide/overview.md [hunk 4]
```diff
@@ -1,5 +1,139 @@
...
+connector](https://github.com/infinyon/fluvio-connectors/blob/c674c960cb3ddef265c7ff34afc0ec8bfc4adb47/rust-connectors/sources/http/Makefile#L1-L4).
+
+Our continuous integration will run `make -C ./path-to-your-connector/Makefile
+test` on each pull request.
+
+
+## 5. Custom build steps
+
+Should your connector require special build steps such as depending on a C
+static library, we'd ask you to have `build` make rule which handles these
```
Comment (nicholastmosher) [inline content/connectors/developer-guide/overview.md:136]: how about

> we'd ask you to have a `make build` rule

**Review by nicholastmosher — CHANGES_REQUESTED** [review]
Looks good except for a few typos. I was also wondering if you could move `developer-guide/_index.md.hidden` back to `developer-guide/_index.md` so that the proper sidebar header appears.

**Review by nicholastmosher — APPROVED** [review]
Looks good!

**Pull request merged**